xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -701,9 +704,21 @@
  * NOTE: This macro uses variables defined in livetable_filterProperty . It was not meant to be used alone.
  *#
 #macro (livetable_filterDateProperty)
-  ## TODO: The date filter should be smarter than this..
-  #set ($whereSql = "${whereSql} and upper(str(${safe_tableAlias}.value)) like upper(?)")
-  #set ($discard = $whereParams.add("%$filterValue%"))
+  #if ($safe_tableAlias.startsWith('doc.'))
+    #set ($dateProperty = $safe_tableAlias)
+  #else
+    #set ($dateProperty = "${safe_tableAlias}.value")
+  #end
+  #set ($timestamps = $filterValue.split('-'))
+  #if ($filterValue.contains('-') &amp;&amp; $timestamps.size() == 2)
```

Comment:
The first condition is implied by the second so it's enough to use the second condition no?